portal/src/graphql/portal/BotProtectionConfigurationScreen.tsx

```diff
@@ -106,14 +211,8 @@ function constructConfig(
   return produce([config, secrets], ([config, secrets]) => {
     config.bot_protection ??= {};
     config.bot_protection.provider ??= {};
-    config.bot_protection = {
-      enabled: currentState.enabled,
-      provider: {
-        type: currentState.providerType,
-        site_key:
-          currentState.providerConfigs[currentState.providerType]?.siteKey,
-      },
-    };
+    config.bot_protection.requirements ??= {};
+    config.bot_protection = constructBotProtectionConfig(currentState);
```

Comment:
It seems the above ??= in L212-214 has no effect because the final object is from `constructBotProtectionConfig(currentState)`